caasp-multi-master-stack.yaml

```diff
@@ -0,0 +1,406 @@
+heat_template_version: 2017-09-01
+
+description: Template to create a CaaSP cluster on OpenStack
+
+parameter_groups:
+  - label: general
+    description: General Parameters
+    parameters:
+      - image
+      - root_password
+
+  - label: sizing
+    description: Sizing Parameters
+    parameters:
+      - admin_flavor
+      - master_flavor
+      - worker_flavor
+      - worker_count
+      - master_count
+
+  - label: network
+    description: Network Parameters
+    parameters:
+      - external_net
+      - internal_net_cidr
+      - dns_nameserver
+      - master_lb_port_1
+      - master_port_1
+      - master_lb_port_2
+      - master_port_2
+      - master_lb_provider
+
+parameters:
+  image:
+    type: string
+    description: Name of image to use for servers
+    constraints:
+      - custom_constraint: glance.image
+  external_net:
+    type: string
+    description: >
+      Name or ID of public network for which floating IP addresses will be allocated
+    default: floating
+  internal_net_cidr:
+    type: string
+    description: Private network range which servers get deployed. Make sure this doesn't conflict with the any IP addresses CaaSP uses (e.g. the dns_cluster_ip).
+    default: 172.28.0.0/24
+  dns_nameserver:
+    type: string
+    description: Address of a dns nameserver reachable.
+    default: 172.28.0.2
+  admin_flavor:
+    type: string
+    description: Admin Flavor
+    default: m1.large
+    constraints:
+      - custom_constraint: nova.flavor
+  master_flavor:
+    type: string
+    description: Master Flavor
+    default: m1.large
+    constraints:
+      - custom_constraint: nova.flavor
+  worker_flavor:
+    type: string
+    description: Worker Flavor
+    default: m1.large
+    constraints:
+      - custom_constraint: nova.flavor
+  worker_count:
+    type: number
+    description: Number of Worker nodes to boot
+    default: 5
+  master_count:
+    type: number
+    description: Number of Master nodes to boot
+    default: 2
+  master_lb_port_1:
+    type: number
+    description: External load balancer port for master api
+    default: 6443
+  master_port_1:
+    type: number
+    description: Master api port
+    default: 6443
+  master_lb_port_2:
+    type: number
+    description: External load balancer port for master api
+    default: 32000
+  master_port_2:
+    type: number
+    description: Master api port
+    default: 32000
+  master_lb_provider:
+    type: string
+    description: Master api load balancer provider, haproxy or octavia
+    default: octavia
```

Comment:
I would recommend haproxy, then it will work also on older OpenStack deployments